dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -282,7 +282,7 @@ def next_script_level
 
   def user_or_session_level
     if current_user
-      current_user.next_unpassed_progression_level(@script).try(:or_next_progression_level)
```

Comment:
should the old method be deprecated?

Reply:
I wondered about that too, and opened [LP-733](https://codedotorg.atlassian.net/browse/LP-733) to track this clean up. 

We use `next_unpassed_progression_level` in a few other places (home controller, maker controller and a user helper). It's likely that it's causing similar bugs with hidden lessons in those places but I want to investigate a little deeper so I don't break something along the way.  After I check those uses, I'll decide if `next_unpassed_progression` level should be replaced with `next_unpassed_visible_progression_level`, combined with an exclude_hidden param or if we still need it separate for valid use cases.